src/app/components/shell/router.js

```diff
@@ -31,41 +31,66 @@ const Error404 = loadable({
     loader: () => import('~/pages/404/404'),
     loading: () => <h1>404</h1>
 });
-const DefaultLayout = loadable({
-    loader: () => import('~/layouts/default/default'),
-    loading: LoadingPlaceholder
-});
 
-function ImportedPage({name}) {
+function useLoading(name, setLayoutParameters) {
     const {pathname} = useLocation();
-    const Page = React.useMemo(
-        () => {
-            function Loading({error, pastDelay, retry}) {
-                if (error) {
-                    if (error.code === 'MODULE_NOT_FOUND') {
-                        return pathname.endsWith('/') ? <Fallback name={name} />
-                        : <Navigate to={`${pathname}/`} replace />;
-                    }
-                    return <div>Error! <button onClick={ retry }>Retry</button></div>;
-                }
-                if (pastDelay) {
-                    return <LoadingPlaceholder />;
+
+    return React.useCallback(
+        ({error, pastDelay, retry}) => {
+            if (error) {
+                if (error.code === 'MODULE_NOT_FOUND') {
+                    return pathname.endsWith('/')
+                    ? <Fallback name={name} setLayoutParameters={setLayoutParameters} />
+                    : <Navigate to={`${pathname}/`} replace />;
                 }
-                return null;
+                return <div>Error! <button onClick={ retry }>Retry</button></div>;
+            }
+            if (pastDelay) {
+                return <LoadingPlaceholder />;
             }
+            return null;
+        },
+        [name, pathname, setLayoutParameters]
+    );
+}
 
-            return loadable({
-                loader: () => import(`~/pages/${name}/${name}`),
-                loading: Loading,
-                render(loaded, props) {
-                    const Component = loaded.default;
+function usePage(name, setLayoutParameters) {
+    const loading = useLoading(name, setLayoutParameters);
 
-                    return <DefaultLayout><Component {...props} /></DefaultLayout>;
-                }
-            });
+    return React.useMemo(
+        () => loadable({
+            loader: () => import(`~/pages/${name}/${name}`),
+            loading,
+            render(loaded, props) {
+                const Component = loaded.default;
+
+                return <Component {...props} />;
+            }
+        }),
+        [name, loading]
+    );
+}
+
+function ImportedPage({name}) {
+    const {pathname} = useLocation();
+    const [layoutName, setLayoutName] = React.useState('default');
+    const [layoutData, setLayoutData] = React.useState(undefined);
```

Comment:
For non-obvious reasons, it doesn't work to make a single `layoutParameters` state object and access its members for `layoutName` and `layoutData`. It causes an update loop when loading a page with non-default layout (e.g., b2s-fall-24)
Navigating from b2s-fall-24 to the home page via the staxly logo link works as expected. It looks like it might not switch to default if it falls back to a general page, so I have added a line to handle that.

Reply:
It looks like `data` needs a little attention to keep it from having unnecessary reloads, so I tried to optimize that.
This branch is on Dev now. There is still some rebuilding of menus when going between some of the pages (K12 and Webinars under "What we do" for example), but the menus persist for most normal page transitions.